hawkular-inventory-paths/src/main/java/org/hawkular/inventory/paths/Path.java

```diff
@@ -1208,17 +1212,17 @@ public Impl get() {
 
         @SuppressWarnings("unchecked")
         public This resource(String id) {
-            segments.add(new Segment(Resource.class, id));
+            segments.add(new Segment(SegmentType.r, id));
             return (This) this;
         }
 
-        public SDB data(Resources.DataRole role) {
-            segments.add(new Segment(DataEntity.class, role.name()));
+        public SDB data(String role) {
```

Comment:
The nice thing about the previous approach was that you could not construct a path to data with invalid role - say a path to "configurationSchema" under a resource, which is not valid.

Previously that would be impossible, because "configurationSchema" is not in the `Resources.DataRole` enum. With the move to a simple string (I understand your reasons for doing it), this is lost and you can construct invalid paths.

Do you have an idea how we could keep the previous nice behavior?